tests/test_functional_sqlite.py

```diff
@@ -6,16 +9,118 @@
+from geoalchemy2.elements import WKBElement
 from geoalchemy2.elements import WKTElement
 from geoalchemy2.shape import from_shape
+from geoalchemy2.shape import to_shape
 
 from . import select
+from . import skip_case_insensitivity
 from . import test_only_with_dialects
+from .schema_fixtures import TransformedGeometry
+
+
+class TestAdmin:
+    def test_create_drop_tables(
+        self,
+        conn,
+        metadata,
+        Lake,
+        Summit,
+        Ocean,
+        PointZ,
+    ):
+        metadata.drop_all(conn, checkfirst=True)
+        metadata.create_all(conn)
+        metadata.drop_all(conn, checkfirst=True)
+
+    @pytest.mark.parametrize("nullable", [True, False])
+    @pytest.mark.parametrize("level", ["col", "type"])
+    def test_nullable(self, conn, metadata, setup_tables, dialect_name, nullable, level):
+        # Define the table
+        col = Column(
+            "geom",
+            Geometry(
+                geometry_type=None,
+                srid=4326,
+                spatial_index=False,
+                nullable=nullable if level == "type" else True,
+            ),
+            nullable=nullable if level == "col" else True,
+        )
+        t = Table(
+            "nullable_geom_type",
+            metadata,
+            Column("id", Integer, primary_key=True),
+            col,
+        )
+
+        # Create the table
+        t.create(bind=conn)
+
+        elements = []
+        if nullable:
+            elements.append({"geom": None})
+        else:
+            elements.append({"geom": "SRID=4326;LINESTRING(0 0,1 1)"})
+
+        conn.execute(t.insert(), elements)
+
+        if not nullable:
+            with pytest.raises((IntegrityError, OperationalError)):
+                with conn.begin_nested():
+                    conn.execute(t.insert(), [{"geom": None}])
+
+        conn.execute(t.insert(), [{"geom": "SRID=4326;LINESTRING(0 0,1 1)"}])
+
+        results = conn.execute(t.select())
+        rows = results.fetchall()
+
+        assert len(rows) == 2
+
+        # Drop the table
+        t.drop(bind=conn)
+
+    def test_no_geom_type(self, conn):
+        with pytest.warns(UserWarning, match="srid not enforced when geometry_type is None"):
+            # Define the table
+            t = Table(
+                "no_geom_type",
+                MetaData(),
+                Column("id", Integer, primary_key=True),
+                Column("geom", Geometry(geometry_type=None, srid=4326)),
+            )
+
+            # Create the table
+            t.create(bind=conn)
+
+            # Drop the table
+            t.drop(bind=conn)
+
+    def test_explicit_schema(self, conn):
+        # Define the table
+        t = Table(
+            "a_table",
+            MetaData(),
+            Column("id", Integer, primary_key=True),
+            Column("geom", Geometry()),
+            schema="gis",
+        )
+
+        # Create the table
+        t.create(bind=conn)
+
+        # Drop the table
```

Comment:
Is there any check here if the table actually exists and has the correct schema? or is that done implicitly by the `drop()`?

Reply:
The drop only checks that the table exists, but not the schema. I will add a check for that, it will be better indeed.